Add snowflake connector documentation (#7206)
wso2/docs-apim: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md

## What you'll build

This example demonstrates how to use the Snowflake Connector for querying, inserting, batch inserting and deleting data in a Snowflake database. Let's consider a use-case involving a hotel database for managing reservations. Within the hotel database, there exists a table named RESERVATIONS responsible for storing reservation details.

1. Insert a single record to the Snowflake database.
2. Insert multiple records to the Snowflake database.
3. Query data from the Snowflake database.
4. Delete data from the Snowflake database.

All operations are exposed via an API. The API with the context `/snowflakeconnector` has 5 resources.

* `/insertReservation` : Insert a single reservation to the Snowflake database.
* `/insertReservationBatch` : Insert a batch of reservations to the Snowflake database.
* `/getReservationInfo` : Retrieve all the reservations from the Snowflake database.
* `/deleteReservation` : Delete a reservation from the Snowflake database.

## Before you begin

Before you begin, you must have a valid Snowflake account. 

> To use the Snowflake database, you must have a valid Snowflake account. To create a snowflake account, please visit the official Snowflake website and complete the registration process. Once registered you will obtain username and password which you can login to Snowflake account, and the account identifier which is the unique identifier for your Snowflake account within your business entity and the Snowflake network.

Reviewer: ```suggestion
To use the Snowflake database, you must have a valid Snowflake account. To create a snowflake account, please visit the official Snowflake website and complete the registration process. Once registered you will obtain a username and password with which you can log in to your Snowflake account, and the account identifier which is the unique identifier for your Snowflake account within your business entity and the Snowflake network.
```

Author: Fixed.

1. Create a database named `HOTEL_DB` in Snowflake.
2. Select the `PUBLIC` schema and create a table named `RESERVATIONS` with the following columns.

```sql
create table RESERVATIONS (
NICNUMBER String,
FIRSTNAME String,
LASTNAME String,
CHECKIN String,
CHECKOUT String,
ADULTS INT,
CHILDREN INT,
ROOMTYPE String,
SPECIALREQUESTS String
)
```

If you do not want to configure this yourself, you can simply [get the project](#get-the-project) and run it.

## Configure the connector in WSO2 Integration Studio

Connectors can be added to integration flows in [WSO2 Integration Studio](https://wso2.com/integration/integration-studio/). Once added, the operations of the connector can be dragged onto your canvas and added to your sequences.

### Import the connector

Follow these steps to set up the Integration Project and the Connector Exporter Project.

{!includes/reference/connectors/importing-connector-to-integration-studio.md!}

### Add integration logic

> First create a REST API called `SnowflakeConnectorApi` in your project

Reviewer: ```suggestion
First, create a REST API called `SnowflakeConnectorApi` in your project.
```

Author: Fixed

| Name | Context |
| ---------------- | ---------------- |
| SnowflakeConnectorApi  | /snowflakeconnector  |

> Create the following resources in 'SnowflakeConnectorApi' REST API

Reviewer: ```suggestion
Create the following resources in the 'SnowflakeConnectorApi' REST API.
```

Author: Fixed.

| uri-template                |
|-----------------------------| 
| /insertReservation          |
| /getReservationInfo         |
| /insertReservationBatch     |
| /deleteReservation          |

> Let's add the operations to the resources in `SnowflakeConnectorApi` API

Reviewer: ```suggestion
Let's add the operations to the resources in the `SnowflakeConnectorApi` API.
```

Author: Fixed.

#### - /insertReservation

> Users can utilize this resource to insert a single record into the snowflake database. The user will be sending the reservation payload in the request body.

Reviewer: ```suggestion
Users can utilize this resource to insert a single record into the Snowflake database. The user will be sending the reservation payload in the request body.
```

Author: Fixed.

1. In the API insequence drag and drop the [Property Mediator]({{base_path}}/reference/mediators/property-mediator/) to extract the payload from the request body. Let's store the payload in a property named `payload`.
    ```xml
      <property expression="json-eval($)" name="payload" scope="default" type="STRING"/>
   ```

>    2. Drag and drop `execute` operation from **SnowflakeConnector** section.

Reviewer: ```suggestion
   2. Drag and drop the `execute` operation from the **SnowflakeConnector** section.
```

Author: Fixed.

      1. Double click the operation. It will show you the properties section.

>       2. In the General Section of the properties, click on the `+` button next to `Snowflake Connection`

Reviewer: ```suggestion
      2. In the **General** Section of the properties, click on the `+` button next to `Snowflake Connection`.
```

Author: Fixed.

            <payload>{$ctx:payload}</payload>
         </snowflake.batchExecute>
         <respond/>
      </inSequence>
      <outSequence/>
      <faultSequence/>
   </resource>
   <resource methods="DELETE" uri-template="/deleteReservation/{NICNUMBER}">
      <inSequence>
         <property expression="fn:concat('DELETE FROM HOTEL_DB.PUBLIC.RESERVATIONS WHERE NICNUMBER=',get-property('uri.var.NICNUMBER'))" name="deleteQuery" scope="default" type="STRING"/>
         <snowflake.execute configKey="SNOWFLAKE_CONNECTION">
            <executeQuery>{$ctx:deleteQuery}</executeQuery>
         </snowflake.execute>
         <property name="messageType" scope="axis2" type="STRING" value="application/json"/>
         <respond/>
      </inSequence>
      <outSequence/>
      <faultSequence/>
   </resource>
</api>
```

{!includes/reference/connectors/exporting-artifacts.md!}

## Get the project

You can download the ZIP file and extract the contents to get the project code.

<a href="{{base_path}}/assets/attachments/connectors/snowflake-connector.zip">
    <img src="{{base_path}}/assets/img/integrate/connectors/download-zip.png" width="200" alt="Download ZIP">
</a>

!!! tip

> You may need to update the values of Snowflake connection configs before deploying and running this project.

Author: Fixed.

## Deployment

Follow these steps to deploy the exported CApp in the integration runtime.

{!includes/reference/connectors/deploy-capp.md!}

### Testing the resources

Let's test the API. Deploy the carbon application and start the Micro Integrator.

1. Insert Reservation Resource

>     1. Create a file called payload.json with the following payload.

Author: Fixed.
